File: backend/src/repository/dto/create-vex-decision.dto.ts
import { IsString, IsOptional, IsEnum, IsNumber } from 'class-validator';

export class CreateVexDecisionDto {
  @IsString()
  vulnerabilityId: string;

  @IsString()
  packageName: string;

  @IsOptional()
  @IsString()
  purl?: string;

  @IsString()
  @IsEnum(['not_affected', 'affected', 'fixed', 'under_investigation'])
  status: string;

  @IsOptional()
  @IsString()
  justification?: string;

  @IsOptional()
  @IsString()
  response?: string;

  @IsOptional()
  @IsString()
  comment?: string;

  @IsNumber()
  repositoryId: number;
}

File: backend-java/src/main/java/com/asmolabs/zanshin/repository/dto/CreateVexDecisionDto.java
package com.asmolabs.zanshin.repository.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record CreateVexDecisionDto(
    @NotBlank String vulnerabilityId,
    @NotBlank String packageName,
    String purl,
    @NotBlank String status,
    String justification,
    String response,
    String comment,
    @NotNull Long repositoryId
) {}
